microsoft/devhome
# PI: Call Close() on vertical bar window when horizontal window is closed and vice versa

## Detailed description of the pull request / Additional comments
Bug: PI crashes on relaunch when ClipboardMonitoring is enabled and we close the bar from the vertical window.
Problem: When we launch PI, we start both the vertical and horizontal window. As a part of Horizontal window load, we start the ClipboardMonitor instance. Now, when we close vertical window, the horizontal window is still activated and ClipboardMonitor is still running. When we start a new instance of PI, the new horiztontalWindow tries to start ClipboardMonitor again and thus failing at Debug.Assert(ListenerHwnd == listeningWindow, "Why are we trying to start with a different hwnd?");

Fix: Call horizontal window Close() from verticalwindow when closed and vice versa.

Also, updated the ClipboardMonitor to listen on PrimaryWindow HWND instead of barwindow HWND.

## Validation steps performed
1. Launch PI
2. Move to vertical window
3. Close PI
4. Launch PI again from Dev Home Utilities and confirm there is no crash

## PR checklist
- [ ] Closes #3048
